joomla/joomla-cms
# Fix regression in script render (async/defer attributes when directly modifying _scripts property)

Pull Request for New Issue.

### Summary of Changes

If an extension has acessing the `JDocument::_scripts` public property, with the changes in the script render it can now cause the script to add async and defer attributes, even if set to false.

This PR intends to correct that.

### Testing Instructions

1. Code review
2. Add to your template index.php (protostar/isis)
```php
$this->_scripts = array_merge(array('/test3.js' => array('mime' => 'text/javascript', 'defer' => '', 'async' => '')), $this->_scripts);
$this->_scripts = array_merge(array('/test2.js' => array('mime' => 'text/javascript', 'defer' => false, 'async' => false)), $this->_scripts);
$this->_scripts = array_merge(array('/test1.js' => array('mime' => 'text/javascript', 'defer' => true, 'async' => true)), $this->_scripts);
```
- Notice the results are different in 3.6.4 and in 3.7.0 staging (which are incorrect).

```html

<script src="/test1.js" defer async></script>
<script src="/test2.js"></script>
<script src="/test3.js"></script>


<script src="/test1.js" defer async></script>
<script src="/test2.js" defer async></script>
<script src="/test3.js" defer async></script>
```
- Apply patch, notice the results on 3.7.0 staging are now the same as 3.6.4

### Documentation Changes Required

None.
